### PR TITLE
Fixes bug in repeat tests in romanText/translate.py; handles empty measures

### PR DESCRIPTION
**Pull request by malcolmsailor**
While working on addressing #1412 I noticed that there was a syntax error in the romanText in `testRepeats` (my bad!). Each measure number was followed by a colon (i.e., "m1:" rather than "m1"). The first thing this PR fixes is removing those colons.

The colons caused the parser to insert a note in each measure. When I removed them, I discovered that the test was actually failing on measures that are empty except for forward/backward repeat bars. To fix this required two things:
1. explicitly set the offset of backward repeat bars according to the time-signature length
2. if a measure has 0 duration, explicitly set its duration according to the current time-signature

I also added a new test for the copying of empty measures.

### CODE REVIEW

**Comment by mscuthbert** [comment]
Thanks!

File: music21/romanText/translate.py [hunk 1]
```diff
@@ -574,6 +575,10 @@ def translateMeasureLineToken(self, measureLineToken: rtObjects.RTMeasure):
 
         else:
             m = self.translateSingleMeasure(measureLineToken)
+            if self.tsCurrent is not None and m.duration.quarterLength == 0:
+                m.duration = duration.Duration(
+                    self.tsCurrent.barDuration.quarterLength
+                )
```
Comment (mscuthbert) [inline music21/romanText/translate.py:581]: this isn't right, unfortunately.  Duration on a stream is special property that automatically manages its quarterLength based on the duration of elements inside of it.  You'd be repeating an empty measure, not a measure of RN.

What should happen, I think is that the code from 593-605 (was 588-600) from "fillToMeasureToken" should be abstracted out into a new method (`fillMeasureFromPreviousRn(mFill: stream.Measure) -> None`) and then that method should be called instead.  Something needs to go into the repeat measure.  We can't repeat an empty span of time.

File: music21/romanText/translate.py [hunk 2]
```diff
@@ -1661,10 +1682,10 @@ def _test_ending_contents(
...
         empty_bars_with_repeats = textwrap.dedent('''
             Time Signature: 2/4
-            m1: I
-            m2: ||:
-            m3: :||
-            m4: ||: :||
+            m1 I
+            m2 ||:
+            m3 :||
+            m4 ||: :||
```
Comment (mscuthbert) [inline music21/romanText/translate.py:1688]: Add tests that the repeated measures do have an RN.figure == 'I' token in it after doing above.

**Review by mscuthbert — CHANGES_REQUESTED** [review]
Realizing that the empty measures can't end empty (and duration on a stream should not be manipulated) -- proposed solution in comments.

File: music21/romanText/translate.py (hunk 1 above)
Comment (malcolmsailor) [reply music21/romanText/translate.py:581]: OK, I'll do this.

In this connection, it might be worth clarifying the docs on when (if ever) explicitly setting duration is possible:

https://web.mit.edu/music21/doc/moduleReference/moduleStreamBase.html#music21.stream.base.Stream.duration

They describe explicitly setting duration in the way I did above as an "advanced usage". I assumed an "advanced usage" would be ok inside the music21 source which is why I thought it would be ok to do.

Comment (mscuthbert) [reply music21/romanText/translate.py]: Yeah, I'll try to improve the docs for that.

It basically means that no matter how long the duration of the measure is, it'll always report some other number, which is something that an end user might want to do, but which our parsers shouldn't do.

File: music21/romanText/translate.py [hunk 3]
```diff
@@ -590,26 +587,29 @@ def fillToMeasureToken(self, measureToken: rtObjects.RTMeasure):
...
+            self.fillMeasureFromPreviousRn(mFill)
             appendMeasureToRepeatEndingsDict(self.lastMeasureToken,
                                              mFill,
                                              self.repeatEndings, i)
             p.coreAppend(mFill)
         self.lastMeasureNumber = measureToken.number[0] - 1
         self.lastMeasureToken = measureToken
 
+    def fillMeasureFromPreviousRn(self, mFill: stream.Measure) -> None:
+        if self.previousRn is not None:
```
Comment (mscuthbert) [inline music21/romanText/translate.py:599]: this is fine. but there's an easy refactor to avoid nesting.

```
if self.previousRn is None:
    return
```

File: music21/romanText/translate.py [hunk 4]
```diff
@@ -1695,6 +1695,14 @@ def _test_ending_contents(
         _repeat_tester(end_repeat1, 'end', 2.0, 3)
         _repeat_tester(start_repeat2, 'start', 0.0, 4)
         _repeat_tester(end_repeat2, 'end', 2.0, 4)
+        for measure in s[stream.Measure]:
+            rn_iter = measure[roman.RomanNumeral]
+            self.assertEqual(len(rn_iter), 1)
+            # mypy complains about the next line because
+            #   RecursiveIterator.first() has t.Optional type, but we know
+            #   it will not be None because we have just asserted that rn_iter
+            #   has length 1
```
Comment (mscuthbert) [inline music21/romanText/translate.py:1704]: that's fine.  :-)

**Review by mscuthbert — APPROVED** [review]
